### DISCUSSION TITLE
MKS Canable V2.0

### DISCUSSION THREAD

**Discussion by KozhinovAlexander** [CANnectivity/cannectivity#30] [Ideas]
Does anybody works on [MKS Canable v2.0](https://github.com/makerbase-mks/CANable-MKS/blob/main/Hardware%2FMKS%20CANable%20V2.0%2FMKS%20CANable%20V2.0_001%20schematic.pdf) board support for CANnectivity?

**Comment by henrikbrixandersen** [comment]
> Does anybody works on MKS Canable v2.0 board support for CANnectivity?

I am not working on it, but support for more dedicated USB to CAN adapter hardware would be very welcome!

I'd suggest to submit the basic board support to the Zephyr project (as this allows others to use the board definitions for other projects than CANnectivity) - and then submit any CANnectivity-specific configuration/additions here.

On a side note, I am currently working on adding support for for WeAct USB2CANFD V1 (https://github.com/WeActStudio/WeActStudio.USB2CANFDV1).

**Reply by KozhinovAlexander** [reply]
Then I would pick addition of this board to CANnectivity. 
This is very chip one - I paid about 7€ on AliExpress

**Reply by henrikbrixandersen** [reply]
MKS CANable V2.0 board support PR: https://github.com/zephyrproject-rtos/zephyr/pull/81366

**Comment by KozhinovAlexander** [comment]
@henrikbrixandersen FYI: Adding stm32g0 support may pave the way to candlelight original board: https://github.com/linux-automation/candleLightFD

**Reply by henrikbrixandersen** [reply]
> @henrikbrixandersen FYI: Adding stm32g0 support may pave the way to candlelight original board: [linux-automation/candleLightFD](https://github.com/linux-automation/candleLightFD)

Zephyr already supports the STM32G0.

**Reply by KozhinovAlexander** [reply]
This is what we love Zephyr for :)

**Reply by henrikbrixandersen** [reply]
FYI: candleLight board support PR: https://github.com/zephyrproject-rtos/zephyr/pull/80953
